socorro/collector/submitter_app.py

```diff
@@ -313,40 +263,52 @@ def source_iterator(self):
 
     #--------------------------------------------------------------------------
     def _infinite_iterator(self):
+        """the default iterators will yield None objects if there is nothing
+        in the queue.  The submitter doesn't want those, so it declines to
+        to yield them"""
         while True:
-            for crash_id in self.new_crash_source.new_crashes():
-                yield crash_id
-
-    #--------------------------------------------------------------------------
-    def _all_iterator(self):
-        for crash_id in self.new_crash_source.new_crashes():
-            yield crash_id
+            for crash_id in self._basic_iterator():
+                if crash_id is not None:
+                    yield crash_id
 
     #--------------------------------------------------------------------------
     def _limited_iterator(self):
```

Comment:
In both case, the value None is being filtered from the stream of values being yielded from the iterator. A better solution to implement a NoOp filter function in this base class and let the SubmitterApp derived class redefine it to detect ```None```.  That way I don't have to reproduce the iterators at all in the derived class.  It's a win for code-reuse.